### ISSUE TITLE
perhaps SC2140 should be suppressed for "\"" case

### ISSUE THREAD

**Issue by John-L** [koalaman/shellcheck#3507]
#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows a potential problem:

```sh

[sed-test.sh](https://github.com/user-attachments/files/30521071/sed-test.sh)

#!/bin/sh
``sed -rn "s/.*?\ssrc="\""(.*?)"\""\s.*?/\n==>\\1 /gIp" <"jufr/index.html"

#### Here's what shellcheck currently says:
SC2140 (warning)


#### Here's what I wanted to see:
[no error]

**Comment by kurahaupo** [comment]
Question: why write nested double quotes like
 `"Barnie Rubble said "\""Hi Fred"\"" on his way to the quarry"`
when
 `"Barnie Rubble said \"Hi Fred\" on his way to the quarry"`
suffices?

(This is different from _single_ quotes, where `\` is always literal, not an escape of the following character.)

**Comment by wileyhy** [comment]
Well, beginning a sed command with two backticks seems problematic
